test/smoketest_artman.py

```diff
@@ -16,73 +16,75 @@
 
 """Artman smoke tests.
 
-It generates both grpc and gapic client libraries for Google APIs in googleapis
+It generates GAPIC client libraries for Google APIs in googleapis
 repo (except those being blacklisted), and fails if any generation fails.
 The test currently assumes that artman and googleapis folders are under the
 same parent folder.
 """
 
 import argparse
-import fnmatch
+import glob
+import json
 import os
-import re
 import subprocess
 import sys
+import yaml
 
-ARTMAN_CONFIG_BLACKLIST = [
-    'artman_spanner_admin_instance.yaml',
-    'artman_spanner_admin_database.yaml',
-    'artman_bigtable_admin.yaml'
-]
+from google.protobuf import json_format
 
-SUPPORTED_LANGS = ['python', 'java', 'ruby', 'nodejs', 'php', 'go', 'csharp']
+from artman.config.proto import config_pb2
 
+CONFIG_BLACKLIST = [
+    'google/spanner/admin/instance/artman_spanner_admin_instance.yaml',
+    'google/spanner/admin/database/artman_spanner_admin_database.yaml',
+    'google/bigtable/admin/artman_bigtable_admin.yaml'
+]
 
-def run_smoke_test(apis, input_dir):
-    input_dir = os.path.abspath(input_dir)
-    artman_config_dirs = [
-        os.path.join(input_dir, 'gapic', 'api'),
-    ]
 
-    artman_config_whitelist = []
-    if apis:
-        for api in apis.split(','):
-            artman_config_whitelist.append('artman_%s.yaml' % api)
+def run_smoke_test(root_dir):
+    failure = []
+    success = []
+    for artman_yaml_path in glob.glob('%s/google/**/artman_*.yaml' % root_dir,
+                                      recursive=True):
+        if os.path.relpath(artman_yaml_path, root_dir) in CONFIG_BLACKLIST:
+            print('Skip %s' % artman_yaml_path)
```

Comment:
I am suspicious of using `print` in a test, especially since many test frameworks swallow stdout. Where will these `print` messages be captured?

Reply:
Though it is called smoketest, I don't plan to use any test framework. But I would love to follow the standard practice. Shall I use logging then?